test/framework/toolchain.py

```diff
@@ -755,7 +755,7 @@ def test_old_new_iccifort(self):
         self.modtool.purge()
 
         tc = self.get_toolchain('goolfc', version='1.3.12')
-        tc.options.update({'openmp': True})
+        tc.set_options({'openmp': True})
```

Comment:
Well, -fopenmp is not correct for the intel suite, it's -qopenmp in their case.
They still support -fopenmp but it might get deprecated.